g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/pattern/PatternServiceImpl.java

```diff
@@ -207,21 +207,24 @@ private Map<String, Object> getCategories(List<String> categoryList, Map<String,
 		Map<String, Object> resultMap = new LinkedHashMap<>();
 		Map<String, Object> categoriesMap = (Map<String, Object>) patternMap.get(ConstantsKeys.CATEGORIES);
 		Map<String, Object> supplementMap = (Map<String, Object>) categoriesMap.get(ConstantsKeys.SUPPLEMENT);
-		Map<String, Object> suppMap = new HashMap<>();
-		for (String cat : categoryList) {
-			suppMap.put(cat, supplementMap.get(cat));
-		}
 
-		if (categoryList.contains(ConstantsKeys.CURRENCIES) && !categoryList.contains(ConstantsKeys.NUMBERS)) {
+		if (categoryList.contains(ConstantsKeys.CURRENCIES) || categoryList.contains(ConstantsKeys.MEASUREMENTS) || categoryList.contains(ConstantsKeys.DATE_FIELDS) && !categoryList.contains(ConstantsKeys.PLURALS)) {
```

Comment:
Why does currency need plural?

Reply:
currency is a special kind of measurement, it also has single and plural forms, please refer to the cldr data:
https://github.com/unicode-cldr/cldr-numbers-full/blob/32.0.0/main/br/currencies.json